src/lib/viewers/image/ImageBaseViewer.js

```diff
@@ -434,6 +435,42 @@ class ImageBaseViewer extends BaseViewer {
         }
     }
 
+    /**
+     * Prints image using an an iframe.
+     *
+     * @return {void}
+     */
+    print() {
+        const browserName = Browser.getName();
+
+        /**
+         * Called async to ensure resource is loaded for print preview. Then removes listener to prevent
+         * multiple handlers.
+         *
+         * @return {void}
+         */
+        const defaultPrintHandler = () => {
+            if (browserName === 'Explorer' || browserName === 'Edge') {
+                this.printframe.contentWindow.document.execCommand('print', false, null);
+            } else {
+                this.printframe.contentWindow.print();
+            }
+
+            this.printframe.removeEventListener('load', defaultPrintHandler);
+            this.emit('printsuccess');
+        };
+
+        this.printframe = util.openContentInsideIframe(this.imageEl.outerHTML);
+        this.printImages = this.printframe.contentDocument.querySelectorAll('img');
+        this.printImages.forEach(element => {
+            element.setAttribute('style', 'display: block; margin: 0 auto; width: 100%');
+        });
+
+        this.printframe.contentWindow.focus();
+
+        this.printframe.addEventListener('load', defaultPrintHandler);
```

Comment:
I have the same question about the reordering of the function calls -- does it affect the behavior in any way? Is it necessary because of the `querySelector` -> `querySelectorAll` change?

Reply:
Sorry, meant to come back and address this question. I adjusted it because it seems more readable but all in all should act the same way.

Reply:
No worries. That makes sense. Thanks for the explanation!